src/routes/device-logs.ts

```diff
@@ -216,6 +221,13 @@ export const store: RequestHandler = async (req: Request, res: Response) => {
 		const logs: DeviceLog[] = supervisor.convertLogs(ctx, body);
 		if (logs.length) {
 			await getBackend(ctx).publish(ctx, logs);
+			if (LOKI_HOST && LOKI_WRITE_PCT > randomPct()) {
+				try {
+					await loki.publish(ctx, logs);
+				} catch (err) {
+					captureException(err, 'Failed to publish log to Loki');
+				}
+			}
```

Comment:
It's probably worth starting both publish in parallel rather in serial, otherwise we could introduce a bunch of latency to responding which might have a knock on effect

Reply:
Good point, updated